src/main/kotlin/io/zerobase/smarttracing/features/devices/DevicesDao.kt

```diff
@@ -87,4 +89,49 @@ class DevicesDao @Inject constructor(private val graph: GraphTraversalSource) {
             throw EntityCreationException("Error creating scan relationship between devices.", ex)
         }
     }
+
+    fun recordTestResult(testResult: TestResult): ReportId {
+        val reportId = randomUUID().toString()
+        try {
+            val deviceNode = graph.V(testResult.testedParty.value).getIfPresent() ?: throw InvalidIdException(testResult.testedParty)
+            val reporterNode = graph.V(testResult.reportedBy.value).getIfPresent() ?: throw InvalidIdException(testResult.reportedBy)
+            graph.addV("TestResult").
+                property(T.id, reportId).
+                property(single, "verified", testResult.verified).
+                property(single, "testDate", testResult.testDate.toString()).
+                property(single, "result", testResult.result).
+                addE("REPORTED").property(single, "timestamp", testResult.timestamp).from(reporterNode).
+                // go back to the report node to make the second edge
+                inV().addE("REPORT_FOR").to(deviceNode).
+                execute()
+            return ReportId(reportId)
+        } catch (ex: Exception) {
+            log.error("error recording test result. device={} verified={}", testResult.testedParty, testResult.verified, ex)
+            throw EntityCreationException("Error recording test result.", ex)
+        }
+    }
+
+    fun recordSymptoms(data: SymptomSummary): ReportId {
```

Comment:
I don't see data.symptoms being added in here as part of the Symptoms vertex property -- is there a reason for that?

Reply:
because i completely forgot it. 🤦 this is why we have code reviews. good catch